pkg/controller/deployment/deployment_controller_test.go

```diff
@@ -778,18 +773,13 @@ func TestSyncDeploymentCreatesReplicaSet(t *testing.T) {
 
 	d := newDeployment(1, nil)
 	f.dStore = append(f.dStore, d)
+	f.objects = append(f.objects, d)
 
-	// expect that one ReplicaSet with zero replicas is created
-	// then is updated to 1 replica
-	rs := newReplicaSet(d, "deploymentrs-4186632231", 0)
-	updatedRS := newReplicaSet(d, "deploymentrs-4186632231", 1)
-	opt := newListOptions()
+	rs := newReplicaSet(d, "deploymentrs-4186632231", 1)
 
 	f.expectCreateRSAction(rs)
 	f.expectUpdateDeploymentAction(d)
-	f.expectUpdateRSAction(updatedRS)
-	f.expectListPodAction(rs.Namespace, opt)
-	f.expectUpdateDeploymentAction(d)
+	f.expectListPodAction(rs.Namespace, api.ListOptions{})
 
```

Comment:
This is a nice catch. For a moment I thought the old test was correct, but after some debugging I realized you are right. cc @janetkuo (for she maintains the deployment API).

This is a good example of why a minimum registry in the fake clientset is useful, because *i)*syncDeployment() has multiple interactions with the API server and its behavior is driven by the states stored in the API server, and *ii)* we want to verify the sequence of the client requests, which currently cannot be done in an integration test.

An alternative solution is running this as an integration test, and add a wrapper around a real clientset to record all the requests sent. cc @lavalamp.